### ISSUE TITLE
[BUG] Route config with data failing

### ISSUE THREAD

**Issue by ravinsinghd** [compodoc/compodoc#489]
##### **Overview of the issue**
Router with data failing 

`Unhandled Rejection at: Promise {
  <rejected> { SyntaxError: Expected '}' instead of '[' at line 1 column 121 of the JSON5 data. Still to read: "[\"0\"]}]"
    at error (D:\workfolder\ng-doc\node_modules\json5\lib\json5.js:56:25)
    at next (D:\workfolder\ng-doc\node_modules\json5\lib\json5.js:72:17)
    at object (D:\workfolder\ng-doc\node_modules\json5\lib\json5.js:464:25)
    at value (D:\workfolder\ng-doc\node_modules\json5\lib\json5.js:482:20)
    at array (D:\workfolder\ng-doc\node_modules\json5\lib\json5.js:417:36)
    at value (D:\workfolder\ng-doc\node_modules\json5\lib\json5.js:484:20)
    at Object.parse (D:\workfolder\ng-doc\node_modules\json5\lib\json5.js:508:18)
    at loopModulesParser (D:\workfolder\ng-doc\node_modules\@compodoc\compodoc\dist\index-cli.js:881:40)
    at RouterParserUtil.constructRoutesTree (D:\workfolder\ng-doc\node_modules\@compodoc\compodoc\dist\index-cli.js:901:13)
    at Dependencies.getDependencies (D:\workfolder\ng-doc\node_modules\@compodoc\compodoc\dist\index-cli.js:4853:45)
  message: 'Expected \'}\' instead of \'[\' at line 1 column 121 of the JSON5 data. Still to read: "[\\"0\\"]}]"',
  at: 120,
  lineNumber: 1,`

##### **Operating System, Node.js, npm, compodoc version(s)**
Windows 10
Node v8.9.4
Npm 5.6.0
compodoc 1.0.7

##### **Angular configuration, a `package.json` file in the root folder**

Attached [package.txt](https://github.com/compodoc/compodoc/files/1751387/package.txt)

##### **Compodoc installed globally or locally ?**
Both. but run by local npm script

##### **Motivation for or Use Case**
We have array of object where we define the data for router. when we map the object to router data, cli able to build the app properly but compodoc failing to recognise the syntax 

##### **Reproduce the error**
Below repo have code where we can reproduce the issue
https://github.com/ravinsinghd/compodoc-error-reproduce

##### **Suggest a Fix**

In App module see the router config. where we pass the data as object from another variable. this syntax should compile properly

**Comment by vogloblinsky** [comment]
I have implemented a try catch to prevent generation for failing for this kind of routes definition. Coming in 1.0.8 today.

**Comment by arvindr21** [comment]
@vogloblinsky any update on this bug? Routes defined as below are not working either

```js
const routes: Routes = [
  { path: '', component: HomeComponent },
  { path: 'about', component: AboutComponent },
  { path: 'create-thread', component: CreateThreadComponent, canActivate: [AuthGuardLogin] },
  { path: 'view-thread/:id', component: ViewThreadComponent },
  { path: 'edit-thread/:id', component: EditThreadComponent, canActivate: [AuthGuardLogin] },
  { path: 'register', component: RegisterComponent },
  { path: 'login', component: LoginComponent },
  { path: 'logout', component: LogoutComponent },
  { path: 'account', component: AccountComponent, canActivate: [AuthGuardLogin] },
  { path: 'admin', component: AdminComponent, canActivate: [AuthGuardAdmin] },
  { path: 'notfound', component: NotFoundComponent },
  { path: '**', redirectTo: '/notfound' }
];
```

Thanks.

**Comment by manekinekko** [comment]
this bug is now fixed by [v1.0.8](https://github.com/compodoc/compodoc/releases/tag/1.0.8)
